numpy/lib/stride_tricks.py

```diff
@@ -111,6 +111,102 @@ def as_strided(x, shape=None, strides=None, subok=False, writeable=True):
 
     return view
 
+def sliding_window_view(x, shape, step=None, writeable=False):
+    """
+    Create sliding window views of the N dimensions array with the given window
+    shape. Window slides across each dimension of `x` and extract subsets of `x`
+    at any window position.
+
+    For some cases, there may be more efficient approaches.
+
+    Parameters
+    ----------
+    x : ndarray
+        Array to create sliding window views.
+    shape : sequence of int
+        The shape of the window. Must have same length as number of input array dimensions.
+    step: sequence of int, optional
+        The steps of window shifts for each dimension on input array at a time.
+        If given, must have same length as number of input array dimensions.
+        Defaults to 1 on all dimensions.
+    writeable : bool, optional
+        If set to False, the returned array will always be readonly.
+        Otherwise it will be writable if the original array was.
+
+    Returns
+    -------
+    view : ndarray
+        Sliding window views of `x`. view.shape = (x.shape - shape) // step + 1
+
+    Examples
+    --------
+    >>> i, j = np.ogrid[:3,:4]
+    >>> x = 10*i + j
+    >>> shape = (2,2)
+    >>> np.lib.stride_tricks.sliding_window_view(x, shape)
+    array([[[[ 0,  1],
+             [10, 11]],
+
+            [[ 1,  2],
+             [11, 12]],
+
+            [[ 2,  3],
+             [12, 13]]],
+
+
+           [[[10, 11],
+             [20, 21]],
+
+            [[11, 12],
+             [21, 22]],
+
+            [[12, 13],
+             [22, 23]]]])
+
+
+    >>> i, j = np.ogrid[:3,:4]
+    >>> x = 10*i + j
+    >>> shape = (2,2)
+    >>> step = (1,2)
+    >>> sliding_window_view(x, shape, step)
+    array([[[[ 0,  1],
+             [10, 11]],
+
+            [[ 2,  3],
+             [12, 13]]],
+
+
+           [[[10, 11],
+             [20, 21]],
+
+            [[12, 13],
+             [22, 23]]]])
+
+    """
+    shape = np.array(shape, np.int)
+
+    if step is None:
+        step = np.ones(len(x.shape), np.int)
+    else:
+        step = np.array(step, np.int)
+
+    if len(x.shape) != len(shape) or len(shape) != len(step):
+        raise ValueError("window shape or step dimension doesn't match with input array")
+    if np.any(shape <= 0) and np.any(step <= 0):
+        raise ValueError('window shape or step cannot contain non-positive value')
+
+    o = (np.array(x.shape)  - shape) // step + 1 # output shape
+    if np.any(o <= 0):
+        raise ValueError('window shape cannot larger than input array shape')
+
+    strides = x.strides
```

Comment:
I have a bit of a bad code smell considering that `x` is never sanitized?

Reply:
What sanitization are you after?